qrexec/policy/parser.py

```diff
@@ -1651,10 +1693,17 @@ def collect_targets_for_ask(self, request):
                 rule_target = (
                     getattr(rule.action, "target", None) or rule.target
                 )
-                expansion = set(
-                    rule_target.expand(system_info=request.system_info)
-                )
-
+                expansion = set()
+                for potential_target in rule_target.expand(
+                        system_info=request.system_info):
+                    try:
+                        # The policy agent cannot handle UUIDs (and rightly so,
+                        # those are meaningless to humans). Convert them to names.
+                        # FIXME: this causes dispvm name reuse issues, should
```

Comment:
First, does it really? You later translate name back to uuid using `request.system_info`, which was fetched at the request (not agent response) time. But second, now dispvm names are not reused for at least a week, so this comment doesn't really apply.

Reply:
Fixed.